File: Labs/Lab 1/TwoSum.java
import java.util.HashMap;
public class TwoSum {
    public static int[] twosum(int[] nums, int target){
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++){
            if (map.containsKey(target-nums[i]))
                return new int[]{i, map.get(target-nums[i])};
            else {
                map.put(nums[i], i);
            }
        }
        return null;
    }
    public static void main(String[] args){
        int[] answer = twosum(new int[]{1,2,3,4}, 7);
        for (int e: answer){
            System.out.println(e);
        }
    }
}

File: Labs/Lab 1/TwoSum.py
def twosum(nums, target):
    map = {}
    for i in range(len(nums)):
        if target - nums[i] in map:
            return [i, map[target - nums[i]]]
        else:
            map[nums[i]] = i

print(twosum([5,8,2,3], 13))